elide-contrib/elide-dynamic-config-helpers/src/main/resources/elideTableSchema.json

```diff
@@ -0,0 +1,435 @@
+        }
+      },
+      "required": [
+        "name",
+        "type",
+        "definition",
+        "grains"
+      ],
+      "additionalProperties": false
+    }
+  },
+  "type": "object",
+  "properties": {
+    "tables": {
+      "title": "Elide Table Models",
+      "description": "Array of elide Table Models",
+      "type": "array",
+      "uniqueItems": true,
+      "items": {
+        "title": "Elide Table Models",
+        "description": "Array of elide Table Models",
+        "type": "object",
+        "properties": {
+          "name": {
+            "title": "Table Model Name",
+            "description": "The name of the model. This will be the same as the POJO class name.",
+            "type": "string",
+            "pattern": "^[A-Z][A-Za-z]*$"
+          },
+          "schema": {
+            "title": "Table Schema",
+            "description": "The database or schema where the model lives.",
+            "type": "string",
+            "pattern": "^([A-Za-z]*_?[A-Za-z]*)*$",
+            "default": ""
+          },
+          "hidden": {
+            "title": "Hide/Show Table",
+            "description": "Whether this table is exposed via API metadata",
+            "type": "boolean",
+            "default": false
+          },
+          "description": {
+            "title": "Table Model description",
+            "description": "A long description of the model.",
+            "type": "string",
+            "default": "The model name"
+          },
+          "cardinality": {
+            "title": "Model cardinality",
+            "description": "The number of rows in the table: (tiny, small, medium, large, huge). The relative sizes are decided by the table designer(s).",
+            "type": "string",
+            "enum": [
+              "tiny",
+              "small",
+              "medium",
+              "large",
+              "huge"
+            ],
+            "default": "tiny"
+          },
+          "readAccess": {
+            "title": "Table read access",
+            "description": "Read permission for the table.",
+            "type": "string",
+            "default": "Allow All"
+          },
+          "joins": {
+            "title": "Table joins",
+            "description": "Describes SQL joins to other tables for column references.",
+            "type": "array",
+            "items": {
+              "$ref": "#/definitions/join"
+            },
+            "default": []
+          },
+          "measures": {
+            "title": "Table measures",
+            "description": "Zero or more metric definitions.",
+            "type": "array",
+            "items": {
+              "$ref": "#/definitions/measure"
+            },
+            "default": []
+          },
+          "dimensions": {
+            "title": "Table dimensions",
+            "description": "One or more dimension definitions.",
+            "type": "array",
+            "items": {
+              "oneOf": [
```

Comment:
Should this be anyOf?

Reply:
anyOf is for atleast one, since the json/hjson can only match one or the other only one (one of ) works for this case.